bug fix bug mentioned in issue 504

jyizheng opened the pull request:
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

This PR fixes the bug in issue #504.


mysql> select number + 1 as number, number+1 as b  from numbers(1);
+--------+------+
| number |    b    |
+--------+------+
|      1      |    1    |
+--------+------+
1 row in set (0.00 sec)


Expected result:
mysql> select number + 1 as number,  number+1 as b  from numbers(1);

+--------+------+
| number |    b    |
+--------+------+
|      1      |    2     |
+--------+------+

1 row in set (0.00 sec)





## Changelog


- Bug Fix


## Related Issues

Fixes #504 

## Test Plan

Unit Tests

Stateless Tests

BohuTANG commented:
Hi @jyizheng 

Although this pull request is in draft, the title will give an idea of what the draft for, title `bugfix` makes no sense and confusing, it's better to have an illustrative title  here :)

jyizheng commented:
> Hi @jyizheng
> 
> Although this pull request is in draft, the title will give an idea of what the draft for, title `bugfix` makes no sense and confusing, it's better to have an illustrative title here :)

I  see, I have changed the title and added more descriptions for this PR.

jyizheng commented:
@BohuTANG  @zhang2014 @sundy-li  Could you please review this PR?

sundy-li commented on fusequery/query/src/sql/plan_parser.rs:
```diff
@@ -447,6 +448,8 @@ impl PlanParser {
         // In example: Aliases=[("id", (number % 3))]
         let aliases = extract_aliases(&projection_exprs);
 
+        let projection_exprs = RewriteHelper::rewrite_projection_aliases(&projection_exprs)?;
```
Actually, I decide to remove rewrite_projection_aliases, because we choose to mysql alias style rather than clickhouse alias style.

jyizheng replied:
Got it. Should I close this PR?

sundy-li commented:
https://www.db-fiddle.com/f/a5HF97XkNi2DQm4RBpWWzK/0
